assets/wizards/newspack/views/settings/connections/webhooks/endpoint-actions-modals.tsx

```diff
@@ -0,0 +1,424 @@
+							  )
+							: sprintf(
+									/* translators: %s: endpoint title */
+									__( 'Are you sure you want to disable the endpoint %s?', 'newspack-plugin' ),
+									`"${ getDisplayUrl( endpoint.url ) }"`
+							  )
+					}
+					onClose={ () => setAction( null, endpoint.id ) }
+					onConfirm={ () => toggleEndpoint( endpoint ) }
+					disabled={ inFlight }
+				/>
+			) }
+			{ action === 'view' && (
+				<Modal
+					title={ __( 'Latest Requests', 'newspack-plugin' ) }
+					onRequestClose={ () => setAction( null, endpoint.id ) }
+				>
+					<p>
+						{ sprintf(
+							// translators: %s is the endpoint title (shortened URL).
+							__( 'Most recent requests for %s', 'newspack-plugin' ),
+							getEndpointLabel( endpoint )
+						) }
+					</p>
+					{ endpoint.requests.length > 0 ? (
+						<table
+							className={ `newspack-webhooks__requests ${
+								hasEndpointErrors( endpoint ) ? 'has-error' : ''
+							}` }
+						>
+							<tr>
+								<th />
+								<th colSpan={ 2 }>{ __( 'Action', 'newspack-plugin' ) }</th>
+								{ hasEndpointErrors( endpoint ) && (
+									<th colSpan={ 2 }>{ __( 'Error', 'newspack-plugin' ) }</th>
+								) }
+							</tr>
+							{ endpoint.requests.map( request => (
+								<tr key={ request.id }>
+									<td className={ `status status--${ request.status }` }>
+										<Icon icon={ getRequestStatusIcon( request.status ) } />
+									</td>
+									<td className="action-name">{ request.action_name }</td>
+									<td className="scheduled">
+										{ 'pending' === request.status
+											? sprintf(
+													// translators: %s is a human-readable time difference.
+													__( 'sending in %s', 'newspack-plugin' ),
+													moment( parseInt( request.scheduled ) * 1000 ).fromNow( true )
+											  )
+											: sprintf(
+													// translators: %s is a human-readable time difference.
+													__( 'processed %s', 'newspack-plugin' ),
+													moment( parseInt( request.scheduled ) * 1000 ).fromNow()
+											  ) }
+									</td>
+									{ hasEndpointErrors( endpoint ) && (
+										<Fragment>
+											<td className="error">
+												{ request.errors && request.errors.length > 0
+													? request.errors[ request.errors.length - 1 ]
+													: '--' }
+											</td>
+											<td>
+												<span className="error-count">
+													{ sprintf(
+														// translators: %s is the number of errors.
+														__( 'Attempt #%s', 'newspack-plugin' ),
+														request.errors.length
+													) }
+												</span>
+											</td>
+										</Fragment>
+									) }
+								</tr>
+							) ) }
+						</table>
+					) : (
+						<Notice
+							noticeText={ __(
+								"This endpoint hasn't received any requests yet.",
+								'newspack-plugin'
+							) }
+						/>
+					) }
+				</Modal>
+			) }
+			{ [ 'edit', 'new' ].includes( action ?? '' ) && (
+				<Modal
```

Comment:
I think it's worth separating out these different modals into their own components especially considering this file is over 400 lines long. WDYT?

Reply:
It might not be noticeable, but webhooks has already been broken into 7 files from the initial [1 file](https://github.com/Automattic/newspack-plugin/blob/edabd0cc74b23773e88215fed1841ae835cb1750/assets/wizards/connections/views/main/webhooks.js).  Happy to break out further, let me know :)

Reply:
Personally, I don't see a problem with breaking this down even more so long as the result is intuitive and easier to parse. IMO, moving these two modals into their own components will help make this file cleaner, as well as make updating the modals simpler in the future.

That said, I'm not too adamant about this, so happy to leave things as are as well.

Reply:
I'll granularize it further, might result in some ugly prop-drilling (a side-effect of using redux vs context).

Stay tuned, will provide update later today.

Reply:
@chickenn00dle this has been done 2dcee95f11a3e60304e2015851e1433af298555f, let me know if you feel further refinement is necessary?

Reply:
Nice! This LGTM 👍 

> I'll granularize it further, might result in some ugly prop-drilling (part in due to use of redux vs context).

I don't think it's necessary in this case, but as an FYI, you can use redux's `[useSelector](https://react-redux.js.org/api/hooks#useselector)` (which uses context API under the hood) to avoid prop-drilling.

Reply:
> I don't think it's necessary in this case, but as an FYI, you can use redux's [useSelector](https://react-redux.js.org/api/hooks#useselector) (which uses context API under the hood I believe) to avoid prop-drilling.

Thanks for the feedback. I should have been more specific, we're using [WordPress's version of redux](https://developer.wordpress.org/block-editor/reference-guides/packages/packages-data/#createreduxstore). The equivalent is [useSelect](https://developer.wordpress.org/block-editor/reference-guides/packages/packages-data/#useselect), which we are using. 

Also, the need for prop-drilling has nothing to do with Redux (I must have bumped my head when I said that, apologies). What requires the prop drilling on the modal components is the need to use the same instance of `useWizardApiFetch` methods & state throughout the component and it's children.

Reply:
Thanks for clarifying @jaredrethman

> I should have been more specific, we're using [WordPress's version of redux](https://developer.wordpress.org/block-editor/reference-guides/packages/packages-data/#createreduxstore). The equivalent is [useSelect](https://developer.wordpress.org/block-editor/reference-guides/packages/packages-data/#useselect), which we are using.

Ah. That's right.

> the need for prop-drilling has nothing to do with Redux (I must have bumped my head when I said that, apologies). What requires the prop drilling on the modal components is the need to use the same instance of useWizardApiFetch methods & state throughout the component and it's children.

I haven't worked too deeply with `@wordpress/data` for a few years, but IIRC there should be feature parity between it and redux (since I thought it was built on redux?). So it should still be possible to create a new store, and populate a components children with some state without the use of prop-drilling. Again, I don't think that's necessary here though. Was just bringing this up since I saw a mention of "ugly prop-drilling"